clinica/utils/inputs.py

```diff
@@ -217,197 +264,308 @@ def find_sub_ses_pattern_path(
-                     following keys : pattern, description. The optional key is: needed_pipeline
-                             pattern: define the pattern of the final file
-                             description: string to describe what the file is
-                             needed_pipeline (optional): string describing the pipeline(s) needed to obtain the related
-                                                        file
-        raise_exception: if True (normal behavior), an exception is raised if errors happen. If not, we return the file
-                        list as it is
-        n_procs: int, optional
-            Number of cores used to fetch files in parallel (defaults to 1).
-            If set to 1, subjects and sessions will be processed sequentially.
-
-    Returns:
-         list of files respecting the subject/session order provided in input,
-         You should always use clinica_file_reader in the following manner:
+
+    Parameters
+    ----------
+    subjects : List[str]
+        List of subjects.
+
+    sessions : List[str]
+        List of sessions. Must be same size as `subjects` and must correspond.
+
+    input_directory : PathLike
+        Path to the BIDS or CAPS directory to read from.
+
+    information : Dict
+        Dictionary containing all the relevant information to look for the files.
+        The possible keys are:
+
+            - `pattern`: Required. Define the pattern of the final file.
+            - `description`: Required. String to describe what the file is.
+            - `needed_pipeline` : Optional. String describing the pipeline(s)
+              needed to obtain the related file.
+
+    raise_exception : bool, optional
+        If True, an exception is raised if errors happen. If not, we return the file
+        list as it is. Default=True.
+
+    n_procs : int, optional
+        Number of cores used to fetch files in parallel.
+        If set to 1, subjects and sessions will be processed sequentially.
+        Default=1.
+
+    Returns
+    -------
+    results : List[str]
+        List of files respecting the subject/session order provided in input.
+
+    error_message : str
+        Error message which contains all errors encountered while reading the files.
+
+    Raises
+    ------
+    TypeError
+        If `information` is not a dictionary.
+
+    ValueError
+        If `information` is not formatted correctly. See function `_check_information`
+        for more details.
+        If the length of `subjects` is different from the length of `sessions`.
+
+    ClinicaCAPSError or ClinicaBIDSError
+        If multiples files are found for 1 subject/session, or if no file is found.
+
+        .. note::
+            If `raise_exception` is False, no exception is raised.
+
+    Notes
+    -----
+    This function is case insensitive, meaning that the pattern argument can, for example,
+    contain upper case letters that do not exists in the existing file path.
```

Comment:
found by PyCharm's spell checker

```suggestion
    This function is case-insensitive, meaning that the pattern argument can, for example,
    contain upper case letters that do not exist in the existing file path.
```